### PR TITLE
Add instructions for accessing authenticated user info

### PR DESCRIPTION
**Pull request by tarekoraby**

### CODE REVIEW

File: articles/security/enabling-security.adoc [hunk 1]
```diff
@@ -364,6 +365,44 @@ By design, the annotations aren't read from parent layouts or parent views. This
...
+@AnonymousAllowed
+public class MyView extends VerticalLayout {
+
+    public MyView(AuthenticationContext authContext) {
+
+        add(new H2("Everyone can see this"));
+
+        authContext.getAuthenticatedUser(UserDetails.class).ifPresent(user -> {
+            if (user.getAuthorities()
+                    .contains(new SimpleGrantedAuthority("ROLE_ADMIN"))) {
```
Comment (mcollovati) [inline articles/security/enabling-security.adoc:395]: `getAuthorities()` return a collection of `GrantedAuthority`, so maybe better not assume they are `SimpleGrantedAuthority`.
The check could be on `GrantedAuthority.getAuthority()`, something like

```java
user.getAuthorities().stream()
    .anyMatch(grantedAuthority -> "ROLE_ADMIN".equals(grantedAuthority.getAuthority());
```

File: articles/security/enabling-security.adoc [hunk 2]
```diff
@@ -364,6 +365,44 @@ By design, the annotations aren't read from parent layouts or parent views. This
...
+
+@Route(value = "my-view")
+@AnonymousAllowed
+public class MyView extends VerticalLayout {
+
+    public MyView(AuthenticationContext authContext) {
+
+        add(new H2("Everyone can see this"));
+
+        authContext.getAuthenticatedUser(UserDetails.class).ifPresent(user -> {
```
Comment (mcollovati) [inline articles/security/enabling-security.adoc:393]: Maybe worth mentioning that the class used with `getAuthenticatedUser()` should match the type created by the authentication providers used in Spring Security.

**Review by mcollovati — COMMENTED** [review]
Some minor comments, other than that LGTM

File: articles/security/enabling-security.adoc (hunk 1 above)
Comment (tarekoraby) [reply articles/security/enabling-security.adoc]: Updated.

File: articles/security/enabling-security.adoc [hunk 3]
```diff
@@ -364,6 +365,44 @@ By design, the annotations aren't read from parent layouts or parent views. This
...
```
Comment (tarekoraby) [inline articles/security/enabling-security.adoc:392]:
```suggestion
        // Ensure that the class used getAuthenticatedUser() matches the created
        // by the authentication providers used in Spring Security.
        authContext.getAuthenticatedUser(UserDetails.class).ifPresent(user -> {
```

**Review by mcollovati — APPROVED** [review]

**Pull request merged**